sys/crypto/modes/ccm.c

```diff
@@ -105,21 +105,35 @@ int ccm_compute_adata_mac(cipher_t* cipher, uint8_t* auth_data,
     if (auth_data_len > 0) {
         int len;
 
-        /* 16 octet block size + max. 10 len encoding  */
-        uint8_t auth_data_encoded[26], len_encoding = 0;
+        uint8_t auth_data_encoded[16] = {0}, len_encoding = 0;
 
-        if ( auth_data_len < (((uint32_t) 2) << 16)) {       /* length (0x0001 ... 0xFEFF)  */
+        if (auth_data_len < (((uint32_t) 1) << 16) - (1 << 8)) {
```

Comment:
Why not just `(auth_data_len <= 0xFEFF)` ? It would even the need for the comment.

Reply:
+1